### PR TITLE
Make sudo use `decl_error!`

### PR DESCRIPTION
**Pull request by shawntabrizi**
This is a small PR to confirm that I am using `decl_error` correctly here.

It upgrades the Sudo pallet to use `decl_error!`.

### CODE REVIEW

File: frame/sudo/src/lib.rs [hunk 1]
```diff
@@ -122,8 +124,8 @@ decl_module! {
 		#[weight = SimpleDispatchInfo::FreeOperational]
 		fn sudo(origin, proposal: Box<T::Proposal>) {
 			// This is a public call, so we ensure that the origin is some signed account.
-			let sender = ensure_signed(origin)?;
-			ensure!(sender == Self::key(), "only the current sudo key can sudo");
+			let sender = ensure_signed(origin).map_err(|_| "origin must be signed")?;
```
Comment (DarkEld3r) [inline frame/sudo/src/lib.rs]: Maybe `ModuleDispatchError::as_str` should be used here? It gives a less readable error description, but at least the same string won't be used everywhere.

**Comment by shawntabrizi** [comment]
Let's wait until a small update to `decl_error!` make this a little bit better to merge this PR @thiolliere

**Comment by gavofyork** [comment]
@shawntabrizi please reference an issue or PR for what you're waiting.

**Comment by shawntabrizi** [comment]
Updated my comment to include a link to: https://github.com/paritytech/substrate/pull/4392

Basically, we expect every module to use `ensure_signed`, `ensure_root`, or `ensure_none`.

For this one situation, I suggested we look to support `From` trait between `frame_system` and any other pallet.

**Pull request merged**